src/pass.h

```diff
@@ -386,6 +386,15 @@ class Pass {
   // for. This is used to issue a proper warning about that.
   virtual bool invalidatesDWARF() { return false; }
 
+  // Whether this pass modifies Binaryen IR in ways that may require fixups for
+  // the "1a" form of non-nullable locals which is in the wasm spec. If the pass
+  // adds locals not in that form, or moves code around in ways that might break
+  // that validation, this must return true. In that case the pass runner will
+  // automatically run the necessary fixups afterwards.
+  //
+  // For more details see the LocalStructuralDominance class.
+  virtual bool requiresNonNullableLocalFixups() { return true; }
```

Comment:
An alternative would be to default to `false` and depend on the fuzzer to find mistakes. That would make for a more annoying developer experience, but might be worth it to avoid introducing unnecessary performance overhead that would be hard to discover and eliminate.

Reply:
Good question. I'm not sure what's best.

Another option might be to make this pure virtual, so it forces all passes to define it, so we need to think about it at least.

Reply:
I don't have a strong opinion here, and if I weren't thinking about performance so much I would definitely prefer this conservative default. The pure virtual idea seems nice.

Reply:
Thinking on this more, given the overhead is as low as it is, I think the higher priority is to avoid correctness bugs for now. We can look into making this faster later on if that becomes relevant, which might mean flipping the default or going pure virtual.